packages/dev/core/src/Meshes/Node/Blocks/linearClonerBlock.ts

```diff
@@ -0,0 +1,216 @@
+
+        this._vertexData = this.geometry.getConnectedValue(state);
+        state.geometryContext = this._vertexData;
+
+        if (!this._vertexData || !this._vertexData.positions || !this.geometry.isConnected) {
+            state.executionContext = null;
+            state.geometryContext = null;
+            this.output._storedValue = null;
+            return;
+        }
+        const count = this.count.getConnectedValue(state) as number;
+
+        if (count <= 0) {
+            this.output._storedValue = this._vertexData;
+            state.executionContext = null;
+            state.geometryContext = null;
+        }
+
+        const origin = this.origin.getConnectedValue(state) as Vector3;
+        const direction = this.direction.getConnectedValue(state) as Vector3;
+        const transformPerStepOrTotal = this.transformPerStepOrTotal.getConnectedValue(state) as number;
+
+        const rotation = this.rotation.getConnectedValue(state) as Vector3;
+        const rotationPerStepOrTotal = this.rotationPerStepOrTotal.getConnectedValue(state) as number;
+
+        const scale = this.scale.getConnectedValue(state) as Vector3;
+        const scalePerStepOrTotal = this.scalePerStepOrTotal.getConnectedValue(state) as number;
+
+        const countVector = new Vector3(count - 1, count - 1, count - 1);
+        const invertRadians = Math.PI / 180;
+
+        const transformMatrix = Matrix.Identity();
+
+        for (this._currentIndex = 0; this._currentIndex < count; this._currentIndex++) {
+            const clone = this._vertexData.clone();
+
+            let transformOffset = direction.clone().scale(this._currentIndex);
+            if (transformPerStepOrTotal == 1) {
+                const distanceStep = direction.divide(countVector).scale(this._currentIndex);
+                transformOffset = direction.clone().multiply(distanceStep);
+            }
+            transformOffset.addInPlace(origin);
+
+            let rotationOffset = rotation.clone().scale(this._currentIndex);
+            if (rotationPerStepOrTotal == 1) {
+                const rotationStep = rotation.divide(countVector).scale(this._currentIndex);
+                rotationOffset = rotation.clone().multiply(rotationStep);
+            }
+
+            let scaleOffset = scale.clone().scale(this._currentIndex);
+            if (scalePerStepOrTotal == 1) {
+                const scaleStep = scale.divide(countVector).scale(this._currentIndex);
+                scaleOffset = scale.clone().multiply(scaleStep);
+            }
+            scaleOffset.addInPlace(new Vector3(1, 1, 1));
+
+            Matrix.ComposeToRef(
+                scaleOffset,
+                Quaternion.FromEulerAngles(rotationOffset.x * invertRadians, rotationOffset.y * invertRadians, rotationOffset.z * invertRadians),
+                transformOffset,
+                transformMatrix
+            );
+            clone.transform(transformMatrix);
+            this._vertexData.merge(clone);

```

Comment:
to avoid multiple parsing can you use the merge of multiple vertexData at once?
Like her:
https://github.com/BabylonJS/Babylon.js/blob/master/packages/dev/core/src/Meshes/Node/Blocks/mergeGeometryBlock.ts#L108

Reply:
When doing this, it keeps the first vertex data, if I ty to make a new one and just combine the new ones it gives me an error.

What would be the best solution for that?  I guess keep a reference to the first one I clone and modify then merge into that?